hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconStorageContainerManagerFacade.java

```diff
@@ -400,18 +451,200 @@ private void initializeSCMDB() {
+              List<HddsProtos.SCMContainerReplicaProto> containerReplicas
+                  = scmServiceProvider.getContainerReplicas(containerID);
+              boolean isContainerPresentAtRecon =
+                  containers.remove(containerID);
+              if (!isContainerPresentAtRecon) {
+                try {
+                  ContainerWithPipeline containerWithPipeline =
+                      scmServiceProvider.getContainerWithPipeline(containerID);
+                  containerManager.addNewContainer(containerWithPipeline);
+                } catch (IOException e) {
+                  LOG.error("Could not get container with pipeline " +
+                      "for container : {}", containerID);
+                } catch (TimeoutException e) {
+                  LOG.error("Could not add new container {} in Recon " +
+                      "container manager cache.", containerID);
+                }
+              }
+              synchronized (containerInfo) {
+                containerReplicas.forEach(containerReplicaProto -> {
+                  final ContainerReplica replica = buildContainerReplica(
+                      containerInfo, containerReplicaProto);
+                  try {
+                    updateContainerState(containerInfo, replica);
+                    containerManager.updateContainerReplica(
+                        containerInfo.containerID(), replica);
+                  } catch (ContainerNotFoundException e) {
+                    LOG.error("Could not update container replica as " +
+                        "container {} not found.", containerID);
+                  } catch (InvalidStateTransitionException e) {
+                    LOG.error("Invalid state transition for container {}",
+                        containerID);
+                  } catch (IOException e) {
+                    LOG.error("Could not update container {} state.",
+                        containerID);
+                  } catch (TimeoutException e) {
+                    LOG.error("Timeout while updating container {} state.",
+                        containerID);
+                  }
+                });
+              }
+            } catch (IOException e) {
+              LOG.error("Unable to get container replicas for container : {}",
+                  containerInfo.getContainerID());
+            }
+          });
+        } else {
+          LOG.debug("SCM DB sync is already running.");
+          return false;
+        }
       } catch (IOException e) {
         LOG.error("Unable to refresh Recon SCM DB Snapshot. ", e);
+        return false;
       }
-    } else {
-      LOG.error("Null snapshot location got from SCM.");
     }
+    return true;
+  }
+
+  private boolean updateContainerState(ContainerInfo container,
+                                       ContainerReplica replica)
+      throws InvalidStateTransitionException, IOException, TimeoutException {
+    final ContainerID containerId = container.containerID();
```

Comment:
I think we need not update container state based on replica, this will be done automatically by FCR report from DN, task here is to get missing container information which is present at SCM, but not present at DN.

Reply:
We need container replica and state as container health task needs that to process for unhealthy containers. @sumitagrawl

Reply:
Removed call to container replica from SCM.